chart/templates/controller.yaml

```diff
@@ -72,34 +72,49 @@ data:
     #!/bin/env bash
     # Get the in-cluster kubeconfig for KubeFlex Control Planes
     # get-kubeconfig.sh cp_name guess_its_name
+
+    # input parameters
     cp_name="${1%"-system"}" # cp name or cp namespace
-    guess_its_name="$2" # true: try guessing the name of the ibms CP
+    guess_its_name="$2" # true: try guessing the name of the ITS CP
+
+    # check if the CP name is valid or needs to be guessed
     if [ "$cp_name" == "" ] ; then
       if [ "$guess_its_name" == "true" ] ; then
-        for cp in `kubectl get controlplane -o name`; do
-          cp=${cp##*/} # separate just the CP name
-          if [ "$(kubectl get controlplane $cp -o jsonpath="{.metadata.labels['kflex\.kubestellar\.io/cptype']}")" == "its" ] ; then
-            if [ "$cp_name" == "" ] ; then
-              cp_name="$cp"
-            else
-              >&2 echo "ERROR: found more than one Control Plane of type its!"
-              exit 1
+        while [ "$cp_name" == "" ] ; do
+          for cp in `kubectl get controlplane -o name`; do
+            cp=${cp##*/} # separate just the CP name
+            if [ "$(kubectl get controlplane $cp -o jsonpath="{.metadata.labels['kflex\.kubestellar\.io/cptype']}")" == "its" ] ; then

```

Comment:
Why not do `kubectl get controlplane -l 'kflex.kubestellar.io/cptype=its'` to get the list of control planes labeled as ITS?

Reply:
@MikeSpreitzer I reworked the code